### PR TITLE
Display class list

### PR DESCRIPTION
**Pull request by jonathanmcdaniel**
:)

### CODE REVIEW

File: Backend/app.py [hunk 1]
```diff
@@ -145,7 +145,7 @@ def secure():
   logger.info(str(flaskResponse))
   return flaskResponse
 
-@app.route('/class-list/', methods=['GET', 'OPTIONS'])
+@app.route('/user/class/', methods=['GET', 'OPTIONS'])
```
Comment (asp78) [inline Backend/app.py]: route isn't very clear, but we talked about it

File: Backend/mentii/class_ctrl.py [hunk 2]
```diff
@@ -36,6 +21,37 @@ def getActiveClassList(dynamoDBInstance, email=None):
...
+  return response
 
+def getClassCodesFromUser(dynamoDBInstance, email=None):
+  classCodes = []
+  if email is None:
+    email = g.authenticatedUser['email']
+  usersTable = dbUtils.getTable('users', dynamoDBInstance)
+  if usersTable is None:
+    MentiiLogging.getLogger().error("Unable to get users table in getClassCodesFromUser")
+    return None
```
Comment (RyanYoung25) [inline Backend/mentii/class_ctrl.py]: Alex is going to hate this.

File: Backend/mentii/class_ctrl.py [hunk 3]
```diff
@@ -36,6 +21,37 @@ def getActiveClassList(dynamoDBInstance, email=None):
...
+  return classCodes
+
+def getPublicClassList(dynamodb, email=None):
+  response = ControllerResponse()
+  classCodes = getClassCodesFromUser(dynamodb, email)
+  classes = []
+  classesTable = dbUtils.getTable('classes', dynamodb)
+  if classesTable is None:
+    MentiiLogging.getLogger().error("Unable to get classes table in getPublicClassList")
+    return None
```
Comment (RyanYoung25) [inline Backend/mentii/class_ctrl.py]: For this one we should return the response with an error added. That's built into the ControllerResponse object.

File: Backend/app.py [hunk 4]
```diff
@@ -157,6 +157,18 @@ def class_list():
...
   return ResponseCreation.createResponse(res, status)
 
+@app.route('/class/', methods=['GET', 'OPTIONS'])
+@auth.login_required
+def public_class_list():
+  status = 200
+  if request.method =='OPTIONS':
+    return ResponseCreation.createEmptyResponse(status)
+  dynamoDBInstance = getDatabaseClient()
+  res = class_ctrl.getPublicClassList(dynamoDBInstance)
```
Comment (RyanYoung25) [inline Backend/app.py]: The way it's currently written this could throw an exception.

File: Backend/mentii/class_ctrl.py [hunk 5]
```diff
@@ -8,26 +8,11 @@
...
-    classCodes = res['Item']['classCodes']
-
+  classCodes = getClassCodesFromUser(dynamoDBInstance, email)
   #Use the class codes to get the class details for
   # each class.
-  table = dbUtils.getTable('classes', dynamodb)
+  table = dbUtils.getTable('classes', dynamoDBInstance)
   if table is None:
     MentiiLogging.getLogger().error("Unable to get classes table in getActiveClassList")
     return None
```
Comment (asp78) [inline Backend/mentii/class_ctrl.py]: I would change this if block to something of the following
```
if table is None:
  Log shit
  add to response
else
  other stuff
end if

return response
```

ryan mentioned this too

Comment (asp78) [inline Backend/mentii/class_ctrl.py]: aka one return, its an easy job here

File: Backend/mentii/class_ctrl.py (hunk 2 above)
Comment (asp78) [inline Backend/mentii/class_ctrl.py]: Ditto as above

File: Backend/mentii/class_ctrl.py (hunk 3 above)
Comment (asp78) [inline Backend/mentii/class_ctrl.py]: Ditto as above

File: Backend/mentii/class_ctrl.py [hunk 6]
```diff
@@ -36,6 +21,37 @@ def getActiveClassList(dynamoDBInstance, email=None):
...
+def getPublicClassList(dynamodb, email=None):
+  response = ControllerResponse()
+  classCodes = getClassCodesFromUser(dynamodb, email)
+  classes = []
+  classesTable = dbUtils.getTable('classes', dynamodb)
+  if classesTable is None:
+    MentiiLogging.getLogger().error("Unable to get classes table in getPublicClassList")
+    return None
+  res = classesTable.scan()
+  for pclass in res['Items']:
```
Comment (RyanYoung25) [inline Backend/mentii/class_ctrl.py]: There's not error checking on if you were able to get items from the classes table.

File: Backend/tests/test_class_ctrl.py [hunk 7]
```diff
@@ -47,6 +47,19 @@ def test_getActiveClassList(self):
     self.assertNotEqual(response, None)
     print(response)
 
+  def test_getPublicClassList(self):
+    user =  "test@mentii.me"
+    response = class_ctrl.getPublicClassList(dynamodb, user)
+    self.assertNotEqual(response, None)
+    print(response)
+
+def test_getClassCodesFromUser(self):
```
Comment (aaronmsegal) [inline Backend/tests/test_class_ctrl.py]: misaligned

File: Backend/app.py (hunk 4 above)
Comment (jonathanmcdaniel) [inline Backend/app.py]: fixed in class_ctrl

File: Backend/mentii/class_ctrl.py [hunk 8]
```diff
@@ -8,34 +8,52 @@
...
+    #Get the class codes for the user.
     if res is not None and 'Item' in res:
-      classes.append(res['Item'])
+      classCodes = res['Item']['classCodes']
+  return classCodes
 
-  response.addToPayload('classes', classes)
+def getPublicClassList(dynamodb, email=None):
+  response = ControllerResponse()
+  classCodes = getClassCodesFromUser(dynamodb, email)
```
Comment (smc395) [inline Backend/mentii/class_ctrl.py]: We should decide, DB instance first then other parameters or other parameters, DB instance last.

File: Backend/tests/test_class_ctrl.py [hunk 9]
```diff
@@ -47,6 +47,19 @@ def test_getActiveClassList(self):
     self.assertNotEqual(response, None)
     print(response)
 
+  def test_getPublicClassList(self):
+    user =  "test@mentii.me"
+    response = class_ctrl.getPublicClassList(dynamodb, user)
+    self.assertNotEqual(response, None)
+    print(response)
```
Comment (smc395) [inline Backend/tests/test_class_ctrl.py]: do you still want to keep these print statements? Also would write a more detailed test case that tests that the contents of the public class list is equal to something you expect

File: Backend/tests/test_class_ctrl.py [hunk 10]
```diff
@@ -47,6 +47,19 @@ def test_getActiveClassList(self):
...
+
+  def test_getClassCodesFromUser(self):
+    user =  "test@mentii.me"
+    response = class_ctrl.getClassCodesFromUser(dynamodb, user)
+    self.assertNotEqual(response, None)
+    self.assertTrue(len(response) == 3)
+    print(response)
```
Comment (smc395) [inline Backend/tests/test_class_ctrl.py]: keep print statement?

File: Backend/mentii/class_ctrl.py (hunk 8 above)
Comment (jonathanmcdaniel) [inline Backend/mentii/class_ctrl.py]: Changes will be made with alex's db_utils refactor

**Pull request merged**